src/main/java/htsjdk/samtools/CigarOperator.java

```diff
@@ -72,12 +94,18 @@
     public static final CigarOperator HARD_CLIP = H;
     public static final CigarOperator PADDING = P;
 
+
+    CigarOperator(final boolean consumesReadBases, final boolean consumesReferenceBases) {
+        this(consumesReadBases, consumesReferenceBases, null);
+    }
+
     /** Default constructor. */
-    CigarOperator(boolean consumesReadBases, boolean consumesReferenceBases, char character) {
+    CigarOperator(boolean consumesReadBases, boolean consumesReferenceBases, final String string) {
         this.consumesReadBases = consumesReadBases;
         this.consumesReferenceBases = consumesReferenceBases;
-        this.character = (byte) character;
-        this.string = new String(new char[] {character}).intern();
+        this.string = string == null ? name() : string; // since we will provide a constant it is internalized already.
```

Comment:
This is clever but I'd rather just have it be specified in the constructor than have logic for determining it from the enum value name.